JS/Lesson18/onur/recursion.js

```diff
@@ -127,10 +228,64 @@ const nestedObjEx7 = {
 	},
 	h: 5,
 };
+
+function findMaxDepth(obj, depth = 1) {
+	if (typeof obj !== 'object') {
+		return null;
+	}
+
+	for (const prop in obj) {
+		if (typeof obj[prop] === 'object') {
+			return findMaxDepth(obj[prop], depth + 1);
+		}
+	}
+	return depth;
+}
+const findDepth = findMaxDepth(nestedObjEx7);
+console.log("Exercise 7: ", findDepth);
+
+console.log('----------------------------');
 // 8. Write a recursive function to reverse the order of words in a sentence.
 const sentenceEx8 = 'Hello, how are you?';
+
+function toReverseWordsOfString(sentence) {
+	const words = sentence.split(' ');
+	if (sentence.length === 0) {
+		return '';
+	} else {
+		const firstWord = words.shift();
+		const remainWords = toReverseWordsOfString(words.join(' '));
+		if (remainWords === '') {
+			return firstWord;
+		} else {
+			return remainWords + ' ' + firstWord;
+		}
+
+	}
+}
+console.log("Exercise 8: ", toReverseWordsOfString(sentenceEx8));
+
+console.log('----------------------------');
 // 9. Write a recursive function to find the length of the longest word in a sentence.
 const sentenceEx9 = 'The quick brown fox jumps over the lazy dog';
+
+function toFindLongestWord(sentence) {
```

Comment:
Try 'findLengthOfTheLongestWord' (it's good to be very specific here as we are not returning the longest word but the length of it)